presto-main/src/main/java/com/facebook/presto/sql/gen/CursorProcessorCompiler.java

```diff
@@ -265,7 +336,79 @@ private void generateProjectMethod(
                 .ret();
     }
 
-    private static RowExpressionVisitor<BytecodeNode, Scope> fieldReferenceCompiler(Variable cursorVariable)
+    private List<MethodDefinition> generateCommonSubExpressionMethods(
+            Metadata metadata,
+            SqlFunctionProperties sqlFunctionProperties,
+            ClassDefinition classDefinition,
+            CallSiteBinder callSiteBinder,
+            CachedInstanceBinder cachedInstanceBinder,
+            Map<LambdaDefinitionExpression, CompiledLambda> compiledLambdaMap,
+            Map<Integer, Map<RowExpression, VariableReferenceExpression>> commonSubExpressionsByLevel,
+            Map<VariableReferenceExpression, CommonSubExpressionFields> commonSubExpressionFieldsMap)
+    {
+        Parameter properties = arg("properties", SqlFunctionProperties.class);
+        Parameter cursor = arg("cursor", RecordCursor.class);
+        Parameter output = arg("output", BlockBuilder.class);
```

Comment:
What is this `output`? Is this supposed to be the result of the whole expression that will be written to the output page? If that's the case, I don't think the cse functions need it.